src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -327,4 +328,43 @@ public Response bulkRemoveDAAsFromUser(
       return createExceptionResponse(e);
     }
   }
+
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON})
+  @Path("{daaId}/dac/{dacId}")
+  public Response addDacToDaa(
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId,
+      @PathParam("dacId") Integer dacId) {
+    try {
+      dacService.findById(dacId);
+      User user = userService.findUserByEmail(authUser.getEmail());
+      // Assert that the user has the correct DAC permissions to add a DAC to a DAA for the provided DacId.
+      // Admins can add a DAC to a DAA with any DAC, but chairpersons can only add DACs to DAAs for DACs they are a
+      // chairperson for.
+      if (!user.hasUserRole(UserRoles.ADMIN)) {
+        List<Integer> dacIds = user.getRoles().stream().map(UserRole::getDacId).toList();
```

Comment:
Very small performance nit: list contains is less than ideal since it iterates over the whole list. I generally cast to a hash set when calling contains.